extensions-contrib/kafka-emitter/src/main/java/io/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -0,0 +1,185 @@
+  private final AtomicLong invalidLost;
+
+  private final KafkaEmitterConfig config;
+  private final Producer<String, String> producer;
+  private final ObjectMapper jsonMapper;
+  private final MemoryBoundLinkedBlockingQueue<String> metricQueue;
+  private final MemoryBoundLinkedBlockingQueue<String> alertQueue;
+  private final ScheduledExecutorService scheduler;
+
+  public KafkaEmitter(KafkaEmitterConfig config, ObjectMapper jsonMapper) {
+    this.config = config;
+    this.jsonMapper = jsonMapper;
+    this.producer = getKafkaProducer(config);
+    this.metricQueue = new MemoryBoundLinkedBlockingQueue<>(queueMemoryBound);
+    this.alertQueue = new MemoryBoundLinkedBlockingQueue<>(queueMemoryBound);
+    this.scheduler = Executors.newScheduledThreadPool(2);
+    this.metricLost = new AtomicLong(0L);
+    this.alertLost = new AtomicLong(0L);
+    this.invalidLost = new AtomicLong(0L);
+  }
+
+  private Producer<String, String> getKafkaProducer(KafkaEmitterConfig config) {
+    Properties props = new Properties();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, config.getBootstrapServers());
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, DEFAULT_KEY_SERIALIZER);
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, DEFAULT_VALUE_SERIALIZER);
+    props.put(ProducerConfig.RETRIES_CONFIG, DEFAULT_RETRIES);
+    props.putAll(config.getKafkaProducerConfig());
+    queueMemoryBound = (props.containsKey(ProducerConfig.BUFFER_MEMORY_CONFIG) ?
+        Long.parseLong(props.getProperty(ProducerConfig.BUFFER_MEMORY_CONFIG)) : queueMemoryBound);
+
+    return new KafkaProducer<>(props);
+  }
+
+  @Override
+  @LifecycleStart
+  public void start() {
+    scheduler.scheduleWithFixedDelay(new Runnable() {
+      public void run() {
+        sendMetricToKafka();
+      }
+    }, 10, 10, TimeUnit.SECONDS);
+    scheduler.scheduleWithFixedDelay(new Runnable() {
+      public void run() {
+        sendAlertToKafka();
+      }
+    }, 10, 10, TimeUnit.SECONDS);
+    log.info("Starting Kafka Emitter.");
+  }
+
+  private void sendMetricToKafka() {
+    sendToKafka(config.getMetricTopic(), metricQueue);
+  }
+
+  private void sendAlertToKafka() {
+    sendToKafka(config.getAlertTopic(), alertQueue);
+  }
+
+  private void sendToKafka(final String topic, MemoryBoundLinkedBlockingQueue<String> recordQueue) {
+    ObjectContainer<String> objectToSend;
+    try {
+      while((objectToSend = recordQueue.take()) != null) {
+        final ObjectContainer<String> finalObjectToSend = objectToSend;
+        producer.send(new ProducerRecord<String, String>(topic, finalObjectToSend.getData()), new Callback() {
+          @Override
+          public void onCompletion(RecordMetadata recordMetadata, Exception e) {
+            if(e != null) {
+              log.warn(e, "Exception occured!");
+              if(topic.equals(config.getMetricTopic())) {
+                if(!metricQueue.offer(finalObjectToSend)) {
```

Comment:
Why are you offering back the message, I think you should just dropped it if there was an exception sending it. Remember kafka producer inherently has retry mechanism.